crates/egui_dnd/examples/virtual_list.rs

```diff
@@ -0,0 +1,61 @@
+use eframe::egui;
+use eframe::epaint::Margin;
+use egui::{CentralPanel, Frame, Id, ScrollArea};
+use egui_dnd::dnd;
+use egui_virtual_list::VirtualList;
+use rand::prelude::StdRng;
+use rand::{Rng, SeedableRng};
+
+pub fn main() -> eframe::Result<()> {
+    let mut items: Vec<_> = (0..100000).collect();
+    let mut virtual_list = VirtualList::new();
+
+    eframe::run_simple_native(
+        "DnD Virtual List Example",
+        Default::default(),
+        move |ctx, _frame| {
+            CentralPanel::default().show(ctx, |ui| {
+                ScrollArea::vertical().show(ui, |ui| {
+                    let response = dnd(ui, "dnd").show_custom(|ui, iter| {
+                        virtual_list.ui_custom_layout(ui, items.len(), |ui, start_index| {
+                            iter.next(
+                                ui,
+                                Id::new(items[start_index]), // assumes that each integer is unique
+                                start_index,
+                                true,
```

Comment:
Is virtual_list guaranteed to always pass a valid start_index to the layout closure? The call to `InfiniteScroll::ui_custom_layout()` [here](https://github.com/lucasmerlin/hello_egui/blob/3cc6cdc96663704c44bf8afe422ca7cbb9b62241/crates/egui_infinite_scroll/src/lib.rs#L396C1-L403C12), which eventually passes a layout closure into `VirtualList::ui_custom_layout()`[here](https://github.com/lucasmerlin/hello_egui/blob/3cc6cdc96663704c44bf8afe422ca7cbb9b62241/crates/egui_infinite_scroll/src/lib.rs#L300C1-L304C16), checks that the item iterator is valid, but I don't see that check in other closures passed to `VirtualList::ui_custom_layout()`.

Reply:
Yes, start_index should always exist. If the list is empty, the closure will never be called.